experimental/web_dashboard/lib/src/widgets/categories_dropdown.dart

```diff
@@ -0,0 +1,77 @@
+// Copyright 2020, the Flutter project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:web_dashboard/src/api/api.dart';
+
+/// Subscribes to the latest list of categories and allows the user to select
+/// one.
+class CategoryDropdown extends StatefulWidget {
```

Comment:
Moving the subscription stuff means that the parent widget has to manage that, and since this widget is used in a couple of places, I'd like to keep it here.

Reply:
This is the only part of the code I'm not completely clear on. Let's make some time in an upcoming 1:1 to walk through it.

The reason I'm balking is that there's a decent bit of code dedicated to managing the stream subscription and calling `setState`, which is StreamBuilder's purpose in life. That's making me wonder if you can simplify things here by finding a way to use StreamBuilder. To be clear, there may not be, but if there is a way to do it, it's probably a good idea.